### ISSUE TITLE
Feature resuest: remove suspended users from search list

### ISSUE THREAD

**Issue by wmcmps** [danmarsden/moodle-mod_dialogue#129]
Would it be possible to remove suspended students from the search list?
<img width="452" alt="suspended user" src="https://github.com/danmarsden/moodle-mod_dialogue/assets/10821510/42669a0c-68bd-470c-9e01-6297c9dae835">

**Comment by danmarsden** [comment]
sounds like a good idea - if you have internal development capabilities, feel free to submit a pull request to improve this.

**Comment by wmcmps** [comment]
I might give it a try. I would only like to change it on 'Site level'.

Something like this:

<img width="978" alt="dialogue_suspended_users" src="https://github.com/danmarsden/moodle-mod_dialogue/assets/10821510/d4d5af9e-f567-47d5-9341-05f7cd0f7af5">

**Comment by cdipe** [comment]
This would be great @wmcmps !
A suggestion: When not wanting to exclude suspended users (for any reason) it could still be a good thing to have them "marked". Perhaps with some symbol, or just use italics for the name.

**Comment by wmcmps** [comment]
@cdipe: that is a good idea.
Something like this:
![dialogue_suspended_users_2](https://github.com/danmarsden/moodle-mod_dialogue/assets/10821510/4e57f99d-42ce-44f7-bb34-2500af25d74a)
Working on it after hours though. So it will not be ready tomorrow.

**Comment by cdipe** [comment]
@wmcmps yes! Or perhaps just "Show (in italics)" and "Do not show"? The italics wouldn't hurt anyone not needing the info :)

**Comment by annickvdp** [comment]
I removed suspended users in the latest version of the plugin:
in mod/dialogue/classes/local/course_enrolment_manager.php, 
in search_users(),
replace 
` [$enrolledsql, $enrolledparams] = get_enrolled_sql($context, $capability, 0, true);`

by 
`[$enrolledsql, $enrolledparams] = get_enrolled_sql($context, 'mod/dialogue:receive', $groupids, true);`
